sqlmesh/core/context.py

```diff
@@ -1410,23 +1448,26 @@ def _nodes_to_snapshots(nodes: t.Dict[str, Node]) -> t.Dict[str, Snapshot]:
                     cache=fingerprint_cache,
                     ttl=ttl,
                 )
-                snapshots[node.name] = snapshot
+                snapshots[snapshot.name] = snapshot
             return snapshots
 
         snapshots = _nodes_to_snapshots(nodes)
         stored_snapshots = self.state_reader.get_snapshots(snapshots.values())
 
-        unrestorable_snapshot_names = {
-            s.name for s in stored_snapshots.values() if s.name in local_nodes and s.unrestorable
+        unrestorable_snapshots = {
+            snapshot
+            for snapshot in stored_snapshots.values()
+            if snapshot.name in local_nodes and snapshot.unrestorable
         }
-        if unrestorable_snapshot_names:
-            for name in unrestorable_snapshot_names:
-                snapshot_id = snapshots[name].snapshot_id
+        if unrestorable_snapshots:
+            for snapshot in unrestorable_snapshots:
                 logger.info(
-                    "Found a unrestorable snapshot %s. Restamping the model...", snapshot_id
+                    "Found a unrestorable snapshot %s. Restamping the model...", snapshot.name
+                )
+                node = local_nodes[snapshot.name]
+                nodes[snapshot.name] = node.copy(
```

Comment:
That's one reason I thought `Snapshot.fqn` might be helpful - when we mix usage of nodes and snapshots. While reading the code I had to remember the context that snapshot's name is actually fqn now.

Reply:
Yeah that is a positive to it. The negative was that there were two ways to do the same thing. Decided that developers will need to learn that snapshot name is an fqn.